pkg/operator/controller/ingress/controller.go

```diff
@@ -83,9 +85,37 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 	if err := c.Watch(&source.Kind{Type: &iov1.DNSRecord{}}, &handler.EnqueueRequestForOwner{OwnerType: &operatorv1.IngressController{}}); err != nil {
 		return nil, err
 	}
+	if err := c.Watch(&source.Kind{Type: &configv1.Ingress{}}, &handler.EnqueueRequestsFromMapFunc{ToRequests: handler.ToRequestsFunc(reconciler.ingressConfigToIngressController)}, predicate.Funcs{
+		CreateFunc:  func(e event.CreateEvent) bool { return true },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return true },
+		UpdateFunc:  func(e event.UpdateEvent) bool { return true },
+		GenericFunc: func(e event.GenericEvent) bool { return true },
+	}); err != nil {
+		return nil, err
+	}
 	return c, nil
 }
 
+func (r *reconciler) ingressConfigToIngressController(o handler.MapObject) []reconcile.Request {
+	var requests []reconcile.Request
+	controllers := &operatorv1.IngressControllerList{}
+	if err := r.cache.List(context.Background(), controllers, &client.ListOptions{Namespace: o.Meta.GetNamespace()}); err != nil {
```

Comment:
Use `client.InNamespace(r.Namespace)` instead of `&client.ListOptions{Namespace: o.Meta.GetNamespace()}`.